src/common/Configuration/Config.cpp

```diff
@@ -57,6 +58,81 @@ namespace
 
         return true;
     }
+
+    // Converts ini keys to the environment variable key (upper snake case).
+    // Example of conversions:
+    //   SomeConfig => SOME_CONFIG
+    //   myNestedConfig.opt1 => MY_NESTED_CONFIG_OPT_1
+    //   LogDB.Opt.ClearTime => LOG_DB_OPT_CLEAR_TIME
+    std::string IniKeyToEnvVarKey(std::string const& key)
+    {
+        std::string result = "";
+
+        const char *str = key.c_str();
+        int n = key.length();
+
+        char curr          = 0;
+        bool isEnd         = false;
+        bool nextIsUpper   = false;
+        bool currIsNumeric = false;
+        bool nextIsNumeric = false;
+
+        for (int i = 0; i < n; ++i)
+        {
+            curr = str[i];
+            if (curr == ' ' || curr == '.' || curr == '-')
+            {
+                result += '_';
+                continue;
+            }
+
+            isEnd = i == n - 1;
+            if (!isEnd)
+            {
+                nextIsUpper = isupper(str[i + 1]);
+
+                // handle "aB" to "A_B"
+                if (!isupper(curr) && nextIsUpper)
+                {
+                    result += toupper(curr);
```

Comment:
```suggestion
                    result += static_cast<char>(std::toupper(curr));
```
Clang-Tidy: Narrowing conversion from 'int' to signed type 'char'